models/DataObject/AbstractObject.php

```diff
@@ -516,7 +516,17 @@ public static function getTotalCount($config = [])
      */
     protected static function typeMatch(AbstractObject $object)
     {
-        return in_array(static::class, [Concrete::class, __CLASS__], true) || $object instanceof static;
+        if (static::class === Concrete::class && !$object instanceof static) {
+            trigger_deprecation(
+                'pimcore/pimcore',
+                '10.5',
+                'Loading none Concrete objects with the Concrete class will not possible in Pimcore 11'
+            );
+
+            return true;
+        }
+
+        return static::class === __CLASS__ || $object instanceof static;
```

Comment:
What do you think about this? It looks a bit more consistent.

```suggestion
        return static::class === self::class || $object instanceof static;
```